conbench/entities/benchmark_result.py

```diff
@@ -72,26 +84,59 @@ def create(data):
 
         # defaults
         benchmark_result_data = {}
-        complete_iterations = False
+
+        # Note(JP): if no data was provided, should this stay `True`?
+        # We will see. https://github.com/conbench/conbench/issues/803.
+        any_errored_iteration = True
 
         if has_stats:
-            benchmark_result_data = data["stats"]
+            # Note(JP): this is the result of marshmallow
+            # deserializaiton/validation against the `BenchmarkResultCreate`
+            # schema below. This needs better naming. Most importantly, we
+            # should make use of typing information derived from the schema.
+            benchmark_result_data = data["stats"].copy()
+
+            # Note(JP): I think we may want to emit a Bad Request response when
+            # benchmark_result_data['data'] does not contain any non-null
+            # value. An empty list would then also be rejected right away. `if
+            # benchmark_result_data.get("data")` is probably supposed to catch
+            # the case where an empty list was provided.
 
             # calculate any missing stats if data available
-            if data["stats"].get("data"):
-                dat = [to_float(x) for x in benchmark_result_data["data"]]
```

Comment:
Before, it was practically impossible to understand what `dat` can become. 

I therefore added a code comment and started using the unambiguous built-in `float()`, `None`, and `math.nan`. 

The IDE still does not know what `x` can be. This will be an important future improvement.